zerver/data_import/slack.py

```diff
@@ -1296,13 +1329,28 @@ def fetch_shared_channel_users(
 
     all_messages = get_messages_iterator(slack_data_dir, added_channels, {}, {})
     for message in all_messages:
-        user_id = get_message_sending_user(message)
-        if user_id is None or user_id in normal_user_ids:
-            continue
-        mirror_dummy_user_ids.add(user_id)
+        if is_integration_bot_message(message):
+            # This message is likely from an integration bot. Since Slack's integration
+            # bots doesn't have user profiles, we need to artificially create users for
+            # them to convert their messages.
+            bot_id = message["bot_id"]
+            if bot_id in integration_bot_users:
+                continue
+            bot_info = get_slack_api_data(
+                "https://slack.com/api/bots.info", "bot", token=token, bot=bot_id
+            )
+            bot_user = convert_bot_info_to_slack_user(bot_info)
+
+            user_list.append(bot_user)
+            integration_bot_users.append(bot_id)
+        else:
+            user_id = get_message_sending_user(message)
+            if user_id is None or user_id in normal_user_ids:
+                continue
+            mirror_dummy_user_ids.add(user_id)
 
     # Fetch data on the mirror_dummy_user_ids from the Slack API (it's
-    # not included in the data export file).
+    # not included in the data export file), except for integration bot users.
```

Comment:
I don't understand this comment change; `mirror_dummy_user_ids` doesn't contain the integration bot users, so it seems correct as previously written?

Reply:
Ah I missed this one. Yes `mirror_dummy_user_ids` don't have Slack's integration bot users.